### DISCUSSION TITLE
How to rename a single coordinate in a DataArray?

### DISCUSSION THREAD

**Discussion by jaxengodfrey** [pydata/xarray#9097] [Q&A]
Hello all,

I'm trying to use `xarray.DataArray.rename` to rename a specific coordinate of an existing DataArray. The documentation page says to pass "a mapping from old names to new names for coordinates or dimensions". However, when I do this for a specific coordinate value, rather than a dimension, it throws an error. Bellow is an example of what I'm attempting to do:

```python
import xarray as xr
import numpy as np

arr = xr.DataArray(np.zeros((3,10)), dims = ['x', 'y'], coords = {'x': ['a', 'b', 'c'], 'y': np.arange(10)})
arr.rename(a = 'A')
```
which throws this error:
```python
ValueError: cannot rename 'a' because it is not a variable or dimension in this dataset
```

I know I could use `assign_coords` instead, but it requires passing in the entire coordinate list, which I'd like to avoid doing since I'm only updating one coordinate. I may also be misunderstanding the terminology, but I believe a "coordinate" refers to a specific value along a given dimension, so based on the doc page for `rename`, what I'm trying to do should work. If anyone could point me in the right direction, I would really appreciate it!

**Comment by dcherian** [comment]
Rename changes the name of the array. I think you want a `replace` which hasn't been implemented yet. There's some discussion and implementation suggestions [here](https://github.com/pydata/xarray/issues/6377)

**Comment by TomNicholas** [comment]
> may also be misunderstanding the terminology, but I believe a "coordinate" refers to a specific value along a given dimension

Not exactly - a "coordinate" refers to the entire `Variable` that stores all the "coordinate values". So in your case the "coordinates" are `x` and `y`. We would normally call what you're referring to a "coordinate value", as in a value of the coordinate array.

**Comment by scottyhq** [comment]
@jaxengodfrey for your example of replacing a specific value along a dimension (`x='a' -> x='A'`) you can do this for the time being: 

 `arr['x']= arr.x.where(arr.x != 'a', 'A')`
